manifests/realm.pp

```diff
@@ -115,30 +115,29 @@
 # Copyright 2013 Patrick Mooney.
 # Copyright (c) IN2P3 Computing Centre, IN2P3, CNRS
 #
-define mit_krb5::realm(
-  $kdc                    = '',
-  $master_kdc             = '',
-  $admin_server           = '',
-  $database_module        = '',
-  $default_domain         = '',
-  $v4_instance_convert    = '',
-  $v4_realm               = '',
-  $auth_to_local_names    = '',
-  $auth_to_local          = '',
-  $kpasswd_server         = '',
-  $v4_realm_convert       = '',
-  $pkinit_anchors         = '',
-  $pkinit_pool            = '',
-  Boolean $rotate_servers = false,
-  $http_anchors           = '',
+define mit_krb5::realm (
+  Optional[Array] $kdc                  = undef,
+  Optional[String] $master_kdc          = undef,
+  Optional[Array] $admin_server         = undef,
```

Comment:
I don't think this should be an `Array`. Even if the current templating code is considering this as an `array_fields`.
`krb5.conf(5)` states that:
1. `admin_server` should point to the "_primary Kerberos server_" (singular)
2.  In the `kpasswd_server` section, "_Finally, port 464 on the admin_server host will be tried_" . Still a single host is expected here.

I think we should go for a `String` here.